server/routes/users.js

```diff
@@ -0,0 +1,116 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+// Input validation
+const validateRegister = require("../user-validation/register");
+const validateLogin = require("../user-validation/login");
+
+const User = require("../models/User");
+
+// POST /users/register
+router.post("/register", (req, res) => {
+  const { errors, isValid } = validateRegister(req.body);
+
+  //validate registration
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+
+  // Check if email is in db
+  User.findOne({ email: req.body.email }).then((user) => {
+    if (user) {
+      return res.status(400).json({ email: "Email already exists" });
+    } else {
+      const newUser = new User({
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        email: req.body.email,
+        password: req.body.password,
+      });
+
+      // Encrypt password before saving in db
+      bcrypt.genSalt(10, (err, salt) => {
+        bcrypt.hash(newUser.password, salt, (err, hash) => {
```

Comment:
I'd prefer if some of this logic was actually in the model file

Reply:
Refactored the code to follow MVC. Encrypting is done on the model now